### PR TITLE
LieAlgebras: Introduce `WeightLattice`

### PR DESCRIPTION
**Pull request by lgoettgens**
Changes:
- Introduce `WeightLattice`, a formal parent for `WeightLatticeElem`
- `character` and friends now return a `Dict{WeightLatticeElem, T}` (with default `T = Int`)
- Weyl actions: remove some duplicate ugliness; instead add a warning to the docs page

preview: https://docs.oscar-system.org/previews/PR4344/Experimental/LieAlgebras/

### CODE REVIEW

**Review by felix-roehrich — APPROVED** [review]

**Comment by lgoettgens** [comment]
can we get this merged? @fingolfin or someone else?

**Comment by lgoettgens** [comment]
Re-drafting until conflicts with https://github.com/oscar-system/Oscar.jl/pull/4339 are resolved.

**Comment by lgoettgens** [comment]
CI is happy with my merge resolution. So this is again ready for merge

File: experimental/LieAlgebras/docs/src/weyl_groups.md [hunk 1]
```diff
@@ -7,6 +7,12 @@ DocTestSetup = Oscar.doctestsetup()
 
 Weyl groups are represented by objects of type `WeylGroup <: Group`, and their elements by `WeylGroupElem <: GroupElement`.
 
+!!! warning
+    Weyl groups in OSCAR afford both left and right actions on roots and weights.
+    Note however, that **the left action** is the default (to align with the literature),
+    and all more complex functionality is defined with respect to the left action, e.g.
+    [`conjugate_dominant_weight_with_elem(::WeightLatticeElem)`](@ref).
```
Comment (joschmitt) [inline experimental/LieAlgebras/docs/src/weyl_groups.md:14]: This is none of my business, but I spent too much time transposing-matrices-until-the-result-looks-correct to not comment here:
So far, my understanding was that all group actions are from the right in OSCAR (at least by default) and we happily break textbook conventions at least in the linear algebra and invariant theory to have this consistent. Why not here?

Comment (lgoettgens) [reply experimental/LieAlgebras/docs/src/weyl_groups.md:14]: Short answer: this is still in discussion with Max (before this migrates to `src/`). This is the current state of things, and I wanted to document how it is *right now* (but subject to change). The removal of some `*_left_*` and `*_right_*` functions in this PR is to clean things up that partially made it possible to do both action directions, but this didn't seem like a healthy way forward (and it didn't think of all places where this was an issue). So I would like to clean this up again, further discuss with Max and afterwards tackle this more systematically.

Comment (joschmitt) [reply experimental/LieAlgebras/docs/src/weyl_groups.md:14]: Okay, thanks!

**Review by joschmitt — APPROVED** [review]
The Aachen people seem to be happy with it.

**Pull request merged**